src/react-components/profile-entry-panel.js

```diff
@@ -62,28 +82,27 @@ class ProfileEntryPanel extends Component {
     e.stopPropagation();
   };
 
-  setAvatarStateFromIframeMessage = e => {
-    if (e.source !== this.avatarSelector.contentWindow) {
-      return;
-    }
-    this.setState({ avatarId: e.data.avatarId });
-
-    // Send it back down to cause avatar picker UI to update.
-    this.sendAvatarStateToIframe();
-  };
-
-  sendAvatarStateToIframe = () => {
-    this.avatarSelector.contentWindow.postMessage({ avatarId: this.state.avatarId }, location.origin);
+  setAvatarFromMediaResult = ({ detail: entry }) => {
+    this.setState({
+      avatarId: entry.id,
+      avatarType: getAvatarType(entry.id),
+      avatarGltfUrl: entry.gltfs.avatar
+    });
+    // Replace history state with the current avatar id since this component gets destroyed when we open the
+    // avatar editor and we want the back button to work. We read the history state back via the avatarId prop.
+    // We read the current state key from history since it could be "overlay" or "entry_step".
+    replaceHistoryState(this.props.history, this.props.history.location.state.key, "profile", { avatarId: entry.id });
   };
 
-  componentDidMount() {
+  async componentDidMount() {
+    this.setState({ avatarGltfUrl: await getAvatarGltfUrl(this.state.avatarId) });
     if (this.nameInput) {
       // stop propagation so that avatar doesn't move when wasd'ing during text input.
       this.nameInput.addEventListener("keydown", this.stopPropagation);
       this.nameInput.addEventListener("keypress", this.stopPropagation);
       this.nameInput.addEventListener("keyup", this.stopPropagation);
     }
-    window.addEventListener("message", this.setAvatarStateFromIframeMessage);
+    this.scene.addEventListener("action_selected_media_result_entry", this.setAvatarFromMediaResult);
```

Comment:
Would have been nice for this to be an `onItemSelected` prop for the mediaBrowser, but maybe is more work than its worth.

Reply:
On second thought that does't really work since the media browser is invoked by url not actually embedded as a component.

Reply:
Yeah, it's not great. Might be worth rethinking how the whole media browser works if we keep running into problems with it.